flytekit/common/tasks/notebook.py

```diff
@@ -0,0 +1,542 @@
+                k: _type_helpers.get_sdk_type_from_literal_type(v.type)
+                for k, v in _six.iteritems(self.interface.inputs)
+            })
+        )
+
+    @_exception_scopes.system_entry_point
+    def execute(self, context, inputs):
+        """
+        :param flytekit.engines.common.EngineContext context:
+        :param flytekit.models.literals.LiteralMap inputs:
+        :rtype: dict[Text, flytekit.models.common.FlyteIdlEntity]
+        :returns: This function must return a dictionary mapping 'filenames' to Flyte Interface Entities.  These
+            entities will be used by the engine to pass data from node to node, populate metadata, etc. etc..  Each
+            engine will have different behavior.  For instance, the Flyte engine will upload the entities to a remote
+            working directory (with the names provided), which will in turn allow Flyte Propeller to push along the
+            workflow.  Where as local engine will merely feed the outputs directly into the next node.
+        """
+        inputs_dict = _type_helpers.unpack_literal_map_to_sdk_python_std(inputs, {
+            k: _type_helpers.get_sdk_type_from_literal_type(v.type) for k, v in _six.iteritems(self.interface.inputs)
+        })
+
+        # Execute Notebook via Papermill.
+        input_notebook_path = self._notebook_path
+        output_notebook_path = input_notebook_path[:len(input_notebook_path) - 6] + '-out' + input_notebook_path[len(
+            input_notebook_path) - 6:]
+        pm.execute_notebook(
+            input_notebook_path,
+            output_notebook_path,
+            parameters=inputs_dict
+        )
+
+        # Parse Outputs from Notebook.
+        outputs = None
+        with open(output_notebook_path) as json_file:
+            data = json.load(json_file)
+            for p in data['cells']:
+                meta = p['metadata']
+                if "outputs" in meta["tags"]:
+                    outputs = ' '.join(p['outputs'][0]['data']['text/plain'])
+
+        if outputs is not None:
+            dict = _literal_models._literals_pb2.LiteralMap()
+            text_format.Parse(outputs, dict)
+
+        # Add output_notebook as an output to the task.
+        output_notebook = _task_output.OutputReference(
+            _type_helpers.get_sdk_type_from_literal_type(Types.Blob.to_flyte_literal_type()))
+        output_notebook.set(output_notebook_path)
+
+        output_literal_map = _literal_models.LiteralMap.from_flyte_idl(dict)
+        output_literal_map.literals[OUTPUT_NOTEBOOK] = output_notebook.sdk_value
+
+        return {
+            _constants.OUTPUT_FILE_NAME: output_literal_map
+        }
+
+    @property
+    def container(self):
+        """
+        If not None, the target of execution should be a container.
+        :rtype: Container
+        """
+
+        # Find task_name
+        task_module = _importlib.import_module(self.instantiated_in)
```

Comment:
Can we use this to resolve relative paths too?

`os.path.join(os.path.dirname(task_module.__file__), user_specified_path)`

Reply:
done